pkg/handler/handler.go

```diff
@@ -33,22 +65,93 @@ func New(b *storage.BucketHandle, cacheDir, listenAddress string) *http.Server {
 	}
 }
 
-func (pbh *prowBucketHandler) router(resp http.ResponseWriter, r *http.Request) {
-	if strings.HasPrefix(r.URL.String(), "/logs") {
-		pbh.handleLogRequest(resp, r)
+func (pbh *prowBucketHandler) router(resp http.ResponseWriter, req *http.Request) {
+	pbh.requestedURL = req.URL.Path
+
+	if strings.HasPrefix(req.URL.String(), "/logs") {
+		pbh.handleLogRequest(resp, req)
+		return
+	}
+
+	if req.URL.Path != "/" {
+		http.NotFound(resp, req)
 		return
 	}
-	resp.WriteHeader(http.StatusNotFound)
+
+	if !isAuthenticated(req) {
+		page, _ := ioutil.ReadFile("pkg/handler/login.html")
+		fmt.Fprintf(resp, string(page))
+	}
 }
 
-func (pbh *prowBucketHandler) handleLogRequest(resp http.ResponseWriter, r *http.Request) {
-	log.Printf("Got request for %s", r.URL.Path)
-	if !pathChecker.MatchString(r.URL.Path) {
+func (pbh *prowBucketHandler) handleLogRequest(resp http.ResponseWriter, req *http.Request) {
+	log.Printf("Got request for %s", req.URL.Path)
+	if !pathChecker.MatchString(req.URL.Path) {
 		resp.WriteHeader(http.StatusForbidden)
 		return
 	}
 
-	bucketPath := strings.Replace(r.URL.Path, "/logs", "pr-logs/pull", 1)
+	// check if the requested repo is private.
+	// if it is private, point to the login page.
+	match := repoName.FindStringSubmatch(req.URL.Path)
+	var repo string
+	if match != nil && len(match) == 1 {
+		repo = match[0] // will be a string of the form /logs/org_repo
+		repo = strings.Replace(repo[6:], "_", "/", -1)
+	}
+
+	if !contains(pbh.publicRepos, repo) {
+		if !isAuthenticated(req) {
+			page, _ := ioutil.ReadFile("pkg/handler/login.html")
+			fmt.Fprintf(resp, string(page))
+			return
+		}
+	}
+
+	pbh.showLogs(resp, req)
+}
+
+// issueSession issues a cookie session after successful Github login
+func (pbh *prowBucketHandler) issueSession() http.Handler {
+	fn := func(resp http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		githubUser, err := github.UserFromContext(ctx)
+		if err != nil {
+			http.Error(resp, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		isMember, _, err := pbh.githubClient.Organizations.IsMember(ctx, pbh.org, *githubUser.Login)
```

Comment:
Do we want to restrict this only to org members or to members of a particular team? Both are possible. :)